File: app/src/main/java/com/android/ashwiask/tvmaze/shows/ShowDiffUtilItemCallback.java
package com.android.ashwiask.tvmaze.shows;

import android.support.v7.util.DiffUtil;

import com.android.ashwiask.tvmaze.home.Show;

public class ShowDiffUtilItemCallback extends DiffUtil.ItemCallback<Show> {
    @Override
    public boolean areItemsTheSame(Show oldItem, Show newItem) {
        return oldItem.id() == newItem.id();
    }

    @Override
    public boolean areContentsTheSame(Show oldItem, Show newItem) {
        return oldItem.isFavorite() == newItem.isFavorite();
    }
}

File: app/src/main/java/com/android/ashwiask/tvmaze/shows/ShowDiffUtilItemCallback.kt
package com.android.ashwiask.tvmaze.shows

import androidx.recyclerview.widget.DiffUtil
import com.android.ashwiask.tvmaze.network.home.Show

class ShowDiffUtilItemCallback : DiffUtil.ItemCallback<Show>() {
    override fun areItemsTheSame(oldItem: Show, newItem: Show): Boolean {
        return oldItem.id == newItem.id
    }

    override fun areContentsTheSame(oldItem: Show, newItem: Show): Boolean {
        return oldItem.name == newItem.name
    }
}
